src/main/java/org/breedinginsight/api/auth/AuthServiceLoginHandler.java

```diff
@@ -75,10 +83,54 @@ public HttpResponse loginSuccess(UserDetails userDetails, HttpRequest<?> request
         return super.loginFailed(authenticationFailed);
     }
 
+    @Override
+    protected HttpResponse loginSuccessWithCookies(List<Cookie> cookies) {
+        try {
+            String locationUrl = this.jwtCookieConfiguration.getLoginSuccessTargetUrl();
+
+            Optional<HttpRequest<Object>> requestOptional = ServerRequestContext.currentRequest();
+            if (requestOptional.isPresent()){
+                HttpRequest<Object> request = requestOptional.get();
+                if (request.getCookies().contains(loginSuccessUrlCookieName)){
+                    Cookie loginSuccessCookie = request.getCookies().get(loginSuccessUrlCookieName);
+                    String returnUrl = loginSuccessCookie.getValue();
+                    try {
+                        returnUrl = URLDecoder.decode(returnUrl, StandardCharsets.UTF_8.name());
+                        if (isValidURL(returnUrl)){
+                            locationUrl = returnUrl;
+                        }
+                    } catch (UnsupportedEncodingException e){}
+                }
+            }
+
+            URI location = new URI(locationUrl);
+
+            MutableHttpResponse mutableHttpResponse = HttpResponse.seeOther(location);
+
+            Cookie cookie;
+            for(Iterator var4 = cookies.iterator(); var4.hasNext(); mutableHttpResponse = mutableHttpResponse.cookie(cookie)) {
```

Comment:
Curious about these var4 and var6 variable names, maybe could be something a little more informative.

Reply:
This is just copied and pasted from the micronaut loginSuccessCookies method. Not a problem to change them for our use though.

Reply:
changed and pushed